### PR TITLE
Use pydata theme for docs

### PR DESCRIPTION
**Pull request by nvaytet**

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -181,78 +144,71 @@ def add_buttons(
...
+    "secondary_sidebar_items": [],
+    "navbar_persistent": ["search-button"],
+    "show_nav_level": 1,
+    # Adjust this to ensure external links are moved to "Move" menu
+    "header_links_before_dropdown": 6,
+    "pygment_light_style": "github-light-high-contrast",
+    "pygment_dark_style": "github-dark-high-contrast",
+    "logo": {
+        "image_light": "_static/logo-2022.svg",
+        # "image_dark": "_static/logo-dark.svg",
```
Comment (nvaytet) [inline docs/conf.py:159]: Ah, I still need to make a dark logo

File: docs/visualization/index.rst
Comment (jl-wynen) [inline docs/visualization/index.rst:1]: This section is quite short. Does it make sense to move it into the user guide to make the navbar less busy?

**Comment by nvaytet** [comment]
I also need to fix the issue with the tables
Screenshot at 2024-01-18 15-49-18

**Comment by jl-wynen** [comment]
> I also need to fix the issue with the tables Screenshot at 2024-01-18 15-49-18

I didn't see that. What page is this?

**Comment by nvaytet** [comment]
Actually it's only on my local build! Weird...

File: requirements/docs.txt [hunk 2]
```diff
@@ -103,14 +117,14 @@ pillow==10.2.0
     #   matplotlib
 plopp==23.11.0
     # via -r docs.in
-pydata-sphinx-theme==0.8.1
-    # via sphinx-book-theme
+pydata-sphinx-theme==0.15.1
```
Comment (jl-wynen) [inline requirements/docs.txt:120]: Can you update again to 0.15.2? There have been some bugfixes this night.

**Review by jl-wynen — APPROVED** [review]

**Comment by jl-wynen** [comment]
> Actually it's only on my local build! Weird...

Maybe you installed different dependencies locally?

**Pull request merged**